plugins/arcviewer.koplugin/main.lua

```diff
@@ -0,0 +1,188 @@
+        end
+        if name == "" then -- some archivers include subfolder name as a separate entry ending with "/"
+            if path ~= "/" then
+                parse_path(path:sub(1,-2), false) -- filesize == false for subfolders
+            end
+        else
+            if self.list_table[path][name] == nil then
+                self.list_table[path][name] = filesize
+                parse_path(path:sub(1,-2), false) -- go up to include subfolders of the branch into the table
+            end
+        end
+    end
+
+    local std_out = io.popen("unzip ".."-qql \""..file.."\"")
+    for line in std_out:lines() do
+        -- entry datetime not used so far
+        local fsize, fname = string.match(line, "%s+(%d+)%s+%d%d%-%d%d%-%d%d%d%d%s+%d%d:%d%d%s+(.+)")
+        parse_path(fname, fsize or 0)
+    end
+end
+
+function ArcViewer:getItemTable(path)
+    local prefix, item_table
+    if path == nil or path == "" then -- root
+        path = "/"
+        prefix = ""
+        item_table = {}
+    else
+        prefix = path
+        item_table = {
+            {
+                text = BD.mirroredUILayout() and BD.ltr("../ ⬆") or "⬆ ../",
+                path = util.splitFilePathName(path:sub(1,-2)),
+            },
+        }
+    end
+
+    local files, dirs = {}, {}
+    for name, v in pairs(self.list_table[path] or {}) do
+        if v then -- file
+            table.insert(files, {
+                text = name,
+                bidi_wrap_func = BD.filename,
+                path = prefix..name,
+                mandatory = util.getFriendlySize(tonumber(v)),
+            })
+        else -- folder
+            local dirname = name.."/"
+            table.insert(dirs, {
+                text = dirname,
+                bidi_wrap_func = BD.directory,
+                path = prefix..dirname,
+                mandatory = self:getItemDirMandatory(prefix..dirname),
+            })
+        end
+    end
+    local sorting = function(a, b) -- by name, folders first
+        return ffiUtil.strcoll(a.text, b.text)
+    end
+    table.sort(dirs, sorting)
+    table.sort(files, sorting)
+    for _, v in ipairs(dirs) do
+        table.insert(item_table, v)
+    end
+    for _, v in ipairs(files) do
+        table.insert(item_table, v)
+    end
+    return item_table
+end
+
+function ArcViewer:getItemDirMandatory(name)
+    local sub_dirs, dir_files = 0, 0
+    for _, v in pairs(self.list_table[name]) do
+        if v then
+            dir_files = dir_files + 1
+        else
+            sub_dirs = sub_dirs + 1
+        end
+    end
+    local text = T("%1 \u{F016}", dir_files)
+    if sub_dirs > 0 then
+        text = T("%1 \u{F114} ", sub_dirs) .. text
+    end
+    return text
+end
+
+function ArcViewer:extractFile(arcfile, filepath)
+    UIManager:show(ConfirmBox:new{
+        text = _("Extract this file?") .. "\n\n" .. filepath .. "\n\n" ..
+            _("If the file already exists, it will be overwritten."),
+        ok_text = _("Extract"),
```

Comment:
This plugins only extract files ? and we then have to exit it and go browse to the extraction location to then view the file ?
It is a bit lying to its name :) I would be expecting to view an image in ImageViewer, and any other stuff in TextViewer.